hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/crypto/HapiCryptoTransfer.java

```diff
@@ -483,6 +500,42 @@ protected Consumer<TransactionBody.Builder> opBodyDef(final HapiSpec spec) throw
         return builder -> builder.setCryptoTransfer(opBody);
     }
 
+    private void PartitionMoveWithUserSignature(final CryptoTransferTransactionBody.Builder b, final HapiSpec spec) {
+        final var extra = appendedPartitionFromTowithSign.get();
+        final var involved = extra.getLeft();
+        final var fromAccount = TxnUtils.asId(involved[0], spec);
+        final var fromPartitionToken = TxnUtils.asTokenId(involved[1], spec);
+        final var toAccount = TxnUtils.asId(involved[2], spec);
+        final var toPartitionToken = TxnUtils.asTokenId(involved[3], spec);
+        final var amount = extra.getRight();
+        final var reduceUnits = TokenTransferList.newBuilder()
+                .setToken(fromPartitionToken)
+                .addTransfers(
+                        AccountAmount.newBuilder().setAccountID(fromAccount).setAmount(-amount));
+        b.addTokenTransfers(reduceUnits);
+        final var appendUnits = TokenTransferList.newBuilder()
+                .setToken(toPartitionToken)
+                .addTransfers(AccountAmount.newBuilder().setAccountID(toAccount).setAmount(+amount));
+        b.addTokenTransfers(appendUnits);
+    }
+
+    private void PartitionMoveWithoutUserSignature(final CryptoTransferTransactionBody.Builder b, final HapiSpec spec) {
```

Comment:
Other than one line, this code is identical to `PartitionMoveWithUserSignature`. Could consider refactoring common code into a single method to avoid duplication.